### PR TITLE
LOG-15403 Fix connection retry functionality

### PR DESCRIPTION
**Pull request by sbialkowski-r7**
- do not ignore IOException in openConnection, throw it so that reconnect mechanism can work correctly
- few sonar issues

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -217,16 +220,23 @@
...
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-module-junit4</artifactId>
+            <version>${powermock.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito</artifactId>
+            <version>${powermock.version}</version>
             <scope>test</scope>
```
Comment (rjacobs-r7) [inline pom.xml:240]: Good unit tests should not rely on hacks like this

File: src/main/java/com/rapid7/net/AsyncLogger.java [hunk 2]
```diff
@@ -359,6 +360,11 @@ boolean checkValidUUID(String uuid) {
         return true;
     }
 
+    //  VisibleForTesting
```
Comment (rjacobs-r7) [inline src/main/java/com/rapid7/net/AsyncLogger.java:363]:
```suggestion
    @VisibleForTesting
```
?

File: src/main/java/com/rapid7/net/AsyncLogger.java [hunk 3]
```diff
@@ -508,38 +511,33 @@ void dbg(String msg) {
...
+                byte[] temp = header.getBytes(ASCII);
+                this.iopsClient.write(temp, 0, temp.length);
             }
         }
 
         /**
          * Tries to opens connection to InsightOps until it succeeds.
          *
-         * @throws InterruptedException
+         * @throws InterruptedException Thrown when interrupted while sleeping >:(
```
Comment (rjacobs-r7) [inline src/main/java/com/rapid7/net/AsyncLogger.java:533]: You throw these at your alarm clock every morning yeah? :D

File: src/main/java/com/rapid7/net/AsyncLogger.java [hunk 4]
```diff
@@ -555,12 +553,12 @@ void reopenConnection() throws InterruptedException {
...
                 // Wait between connection attempts
-                root_delay *= 2;
-                if (root_delay > MAX_DELAY)
-                    root_delay = MAX_DELAY;
-                int wait_for = root_delay + random.nextInt(root_delay);
-                dbg("Waiting for " + wait_for + "ms");
-                Thread.sleep(wait_for);
+                rootDelay *= 2;
+                if (rootDelay > MAX_DELAY)
+                    rootDelay = MAX_DELAY;
```
Comment (rjacobs-r7) [inline src/main/java/com/rapid7/net/AsyncLogger.java:558]:
```suggestion
                if (rootDelay > MAX_DELAY) {
                    rootDelay = MAX_DELAY;
                }
```

File: pom.xml (hunk 1 above)
Comment (sbialkowski-r7) [reply pom.xml:240]: I mean it's perhaps not ideal, but the alternative is to have worse encapsulation it seems, leaving a lot package private and adding way more constructors

File: src/main/java/com/rapid7/net/AsyncLogger.java (hunk 3 above)
Comment (sbialkowski-r7) [reply src/main/java/com/rapid7/net/AsyncLogger.java:533]: all a thread wants is to sleep peacefully

File: src/main/java/com/rapid7/net/AsyncLogger.java (hunk 2 above)
Comment (sbialkowski-r7) [reply src/main/java/com/rapid7/net/AsyncLogger.java:363]: this is in guava, which isn't here so so I assume we don't want to import the world just for an annotation

File: src/main/java/com/rapid7/net/AsyncLogger.java [hunk 5]
```diff
@@ -555,12 +553,13 @@ void reopenConnection() throws InterruptedException {
                 }
```
Comment (jcolgan-r7) [inline src/main/java/com/rapid7/net/AsyncLogger.java:553]: Is simply catching `IOException` sufficient above? `openConnection()` previously swallowed all exceptions.

Comment (sbialkowski-r7) [reply src/main/java/com/rapid7/net/AsyncLogger.java:553]: should be sufficient.

**Review by jcolgan-r7 — APPROVED** [review]

File: .travis.yml [hunk 6]
```diff
@@ -4,7 +4,7 @@ jdk:
   - openjdk9
   - openjdk10
   - openjdk11
-  - openjdk12
```
Comment (rjacobs-r7) [inline .travis.yml:7]: We're on Java 15 by now. Does this not work beyond Java 11?

File: .java-version [hunk 7]
```diff
@@ -0,0 +1 @@
+15
```
Comment (maquilanti-r7) [inline .java-version:1]: leci will build this using Java 8 so the jar we published can be used as dependency in all projects that use java version >= 8

Comment (sbialkowski-r7) [reply .java-version:1]: so set it to 8?

Comment (maquilanti-r7) [reply .java-version:1]: yes, so we are sure it builds with 8

**Review by maquilanti-r7 — APPROVED** [review]

File: pom.xml [hunk 8]
```diff
@@ -3,8 +3,11 @@
 
     <properties>
         <additionalparam>-Xdoclint:none</additionalparam>
+
         <jackson.version>2.10.1</jackson.version>
         <log4j.version>2.13.3</log4j.version>
+        <powermock.version>1.7.4</powermock.version>
```
Comment (jcolgan-r7) [inline pom.xml:9]: Unused

File: pom.xml [hunk 9]
```diff
@@ -227,7 +231,6 @@
             <groupId>org.mockito</groupId>
             <artifactId>mockito-all</artifactId>
             <version>1.9.5</version>
-            <scope>test</scope>
```
Comment (jcolgan-r7) [inline pom.xml:230]: This should stay

File: pom.xml (hunk 8 above)
Comment (sbialkowski-r7) [reply pom.xml:9]: nice spot

**Review by jcolgan-r7 — APPROVED** [review]

**Review by maquilanti-r7 — APPROVED** [review]

**Pull request merged**